src/scripts/hooks/utils/Ui5ExtensionMocker.ts

```diff
@@ -0,0 +1,30 @@
+interface MockResponse {
+  payload: string;
+  params: {
+    statusCode: number;
+    fetchResponse: boolean;
+  }
+}
+
+export class Ui5ExtensionMocker {
+  private static readonly urlRegexps: RegExp[] = [
```

Comment:
As suggested by Jan, I would generally support mocking each request specifically. With this current solution we can only respond with the same payload for each request.

@PhyNerd what do you think? You mentioned you have mocked one requests specifically returning something instead of 404?